assets/src/design-system/components/input/index.js

```diff
@@ -194,9 +199,10 @@ export const Input = forwardRef(
             onBlur={handleBlur}
             onFocus={handleFocus}
             value={displayedValue}
+            hasSuffix={hasSuffix}
             {...props}
           />
-          {suffix && (
+          {hasSuffix && (
```

Comment:
`suffix=0` would actually have printed `0` to DOM before, as it's falsey but printable.